core/io/linux.rs

```diff
@@ -182,6 +505,23 @@ impl IO for LinuxIO {
     fn get_current_time(&self) -> String {
         chrono::Local::now().format("%Y-%m-%d %H:%M:%S").to_string()
     }
+
+    fn wait_for_completion(&self, timeout: i32) -> Result<()> {
+        let mut inner = self.inner.borrow_mut();
+        let start = std::time::Instant::now();
+        loop {
+            if start.elapsed().as_millis() as i32 >= timeout {
+                log::error!("Timeout waiting for completion {}", timeout);
+                return Err(LimboError::LinuxIOError(
+                    "timeout waiting for completion".into(),
+                ));
+            }
+            if inner.cycle_once()? == SqPushResult::Completed {
```

Comment:
The thought behind this being a new method in the `IO` trait, and not just a loop with a timeout around `self.pager.io.run_once()`, is the caller (btree) with `run_once`, doesn't have the context to know whether all outstanding operations were completed, or just one. So by making this a new method, the logic can be here to break early in that case.

Reply:
Another option I guess would be returning something like `Result<IoStatus>` from `run_once`, which could theoretically be ignored in situations where you only want to cycle it once.

That would allow for the storage/btree `wait_for_completion` to have the same logic.. So it still alters the IO trait.. but it's not adding a new method at least.